core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -229,15 +243,16 @@ object Executor {
         }
 
         def collectMixed() = {
-          val queries   = Vector.newBuilder[(String, ReducedStep[R], FieldInfo)]
-          var names     = List.empty[String]
-          var resolved  = List.empty[ResponseValue]
-          var remaining = steps
-          while (remaining ne Nil) {
+          val queries                       = new VectorBuilder[(String, ReducedStep[R], FieldInfo)]
+          val nil                           = Nil
```

Comment:
Is there some optimization reason to introduce this indirection?

Reply:
This is a small trick that I saw in jsoniter. It stores the Nil object into the stack memory so that it doesn't need to compare it from the heap on each iteration.

It's a very _micro_-optimization that's why I only did it here where we also initialize multiple values with Nil

Reply:
I feel we should comment why we do such trick, otherwise it might get removed by someone else later.

Reply:
"This one weird trick the compiler doesn't want you to know about" 🤡